src/plugins/listops/dictionary_operation.cpp

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2018 Weile Wei
+// Copyright (c) 2018 Parsa Amini
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <phylanx/config.hpp>
+#include <phylanx/ir/dictionary.hpp>
+#include <phylanx/plugins/listops/dictionary_operation.hpp>
+#include <phylanx/util/matrix_iterators.hpp>
+
+#include <hpx/include/lcos.hpp>
+#include <hpx/include/util.hpp>
+#include <hpx/throw_exception.hpp>
+
+#include <cstddef>
+#include <cstdint>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <blaze/Math.h>
+
+///////////////////////////////////////////////////////////////////////////////
+namespace phylanx { namespace execution_tree { namespace primitives
+{
+    ///////////////////////////////////////////////////////////////////////////
+    match_pattern_type const dict_operation::match_data = {
+        hpx::util::make_tuple("dict", std::vector<std::string>{"dict(__1)"},
+            &create_dict_operation, &create_primitive<dict_operation>)};
+
+    ///////////////////////////////////////////////////////////////////////////
+    dict_operation::dict_operation(
+        std::vector<primitive_argument_type>&& operands,
+        std::string const& name, std::string const& codename)
+      : primitive_component_base(std::move(operands), name, codename)
+    {
+    }
+
+    ///////////////////////////////////////////////////////////////////////////
+    primitive_argument_type dict_operation::generate_dict(
+        std::vector<phylanx::ir::range>&& args) const
+    {
+        phylanx::ir::dictionary dict;
+
+        ir::range args_list =
+            extract_list_value_strict(std::move(args[0]), name_, codename_);
```

Comment:
If `dict()` is called without arguments, then `args.size() == 0`. This would lead to an index out of bounds error when accessing `args[0]`. Please add a test for this case.